lint-hrx.ts

```diff
@@ -0,0 +1,161 @@
+import fs from 'fs';
+import p from 'path';
+import yargs from 'yargs/yargs';
+import fileLinesCount from 'file-lines-count';
+
+import {fromPath, SpecDirectory} from './lib-js/spec-directory';
+import RealDirectory from './lib-js/spec-directory/real-directory';
+import VirtualDirectory from './lib-js/spec-directory/virtual-directory';
+import {GithubActionsReporter} from './lib-js/reporter';
+
+/** The number of lines allowed in HRX files. */
+const LINE_THRESHOLD = 500;
+
+/**
+ * Verifies all HRX files in the project `spec` directory are under the
+ * {@link LINE_THRESHOLD}.
+ *
+ * HRX files are unwrapped and transformed into real files when
+ * {@link reporter.fix} is true.
```

Comment:
It seems like a layering mismatch to have the reporter be responsible for tracking whether the linter fixes errors. I know you inherited that from the previous code, but if possible it would be good to fix rather than carrying it forward.

Reply:
given how the linters using the reporter work, the `fix` isn't even needed within the reporter 😱